src/status_im2/common/bottom_sheet_screen/view.cljs

```diff
@@ -0,0 +1,75 @@
+(ns status-im2.common.bottom-sheet-screen.view
+  (:require
+    [react-native.gesture :as gesture]
+    [react-native.navigation :as navigation]
+    [react-native.platform :as platform]
+    [react-native.reanimated :as reanimated]
+    [oops.core :refer [oget]]
+    [status-im2.common.bottom-sheet-screen.style :as style]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [quo.react]
+    [utils.re-frame :as rf]))
+
+(def ^:const drag-threshold 100)
+
+(defn drag-gesture
+  [translate-y opacity scroll-enabled curr-scroll]
+  (->
+    (gesture/gesture-pan)
+    (gesture/on-start (fn [e]
+                        (when (< (oget e "velocityY") 0)
+                          (reset! scroll-enabled true))))
+    (gesture/on-update (fn [e]
+                         (let [translation (oget e "translationY")
+                               progress    (Math/abs (/ translation drag-threshold))]
+                           (when (pos? translation)
+                             (reanimated/set-shared-value translate-y translation)
+                             (reanimated/set-shared-value opacity (- 1 (/ progress 5)))))))
+    (gesture/on-end (fn [e]
+                      (if (> (oget e "translationY") drag-threshold)
+                        (do
+                          (reanimated/set-shared-value opacity (reanimated/with-timing-duration 0 100))
+                          (rf/dispatch [:navigate-back]))
+                        (do
+                          (reanimated/set-shared-value opacity (reanimated/with-timing 1))
+                          (reanimated/set-shared-value translate-y (reanimated/with-timing 0))
+                          (reset! scroll-enabled true)))))
+    (gesture/on-finalize (fn [e]
+                           (when (and (>= (oget e "velocityY") 0)
+                                      (<= @curr-scroll (if platform/ios? -1 0)))
+                             (reset! scroll-enabled false))))))
+
+(defn on-scroll
+  [e curr-scroll]
+  (let [y (oget e "nativeEvent.contentOffset.y")]
+    (reset! curr-scroll y)))
+
+(defn consumer
+  [children skip-background?]
```

Comment:
seems like skip-background? not used

Reply:
It is used line 67. This is an optional second parameter to not have another background opacity when opening 2 bottom-sheets on top of each other. (like opening album-selector on top of photo-selector)

Reply:
mhm that looks complex

Reply:
@flexsurfer you mean the design?